tests/unit/lib/test_opensearch_tls.py

```diff
@@ -1578,7 +1599,10 @@ def test_on_certificate_available_rotation_ongoing_on_this_unit(
         # No action taken, no change on status or certificates
         assert run_cmd.call_count == 0
         assert self.harness.model.unit.status == original_status
-        self.charm.on.certificate_available.defer.called_once()
+        if leader:
+            event_mock.defer.assert_called_once()
+        else:
+            event_mock.defer.assert_not_called()
```

Comment:
I think `event_mock.defer` should rather be: `self.charm.on.certificate_available.defer`

Reply:
I switched the event_mock to the `certificate_available` event since that is what would be the input if the charm calls the callback function.